hack/verify-payload-crds.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
+source "$(dirname "${BASH_SOURCE}")/update-payload-crds.sh"
+
+files=""
+
+# Check there's no diff between the files in their canonical location
+# and the payload-manifests location.
+for f in ${crd_globs}; do
+    files+="$(basename ${f}),"
+    echo "Verifying diff on $(basename $f)"
+    diff "$f" "${SCRIPT_ROOT}/payload-manifests/crds/$(basename "$f")"
+done
+
+# Check that we haven't accidentally added any files that aren't tracked
+# by the crd_globs into the payload CRDs folder.
+for f in "${SCRIPT_ROOT}/payload-manifests/crds/"*; do 
+    if ! echo "${files}" | tr "," "\n" | grep -F -q -x "$(basename "${f}")"; then
+        echo "Found untracked file $(basename $f) in payload CRD manifests, please add the file to the crd_globs in hack/update-payload-crds.yaml"
```

Comment:
I believe the code is technically correct with my previous suggestion, but you could also save some calls to `tr` and `basename`, and reduce the amount of places you need to get quoting right, by adding a couple local variable assignments:
```suggestion
    basename=$(basename "${f}")
    files+=${basename},
    echo "Verifying diff on ${basename}"
    diff "$f" "${SCRIPT_ROOT}/payload-manifests/crds/${basename}"
done

files=$(echo "${files}" | tr "," "\n")

# Check that we haven't accidentally added any files that aren't tracked
# by the crd_globs into the payload CRDs folder.
for f in "${SCRIPT_ROOT}/payload-manifests/crds/"*; do
    basename=$(basename "${f}")
    if ! echo "${files}" | grep -F -q -x "${basename}"; then
        echo "Found untracked file ${basename} in payload CRD manifests.  Please add the file to crd_globs in hack/update-payload-crds.sh."
```

Better yet, could you use `files=${files//,/$'\n'}` instead of `files=$(echo "${files}" | tr "," "\n")`?  That works on my RHEL system with bash 4.4.20, but I haven't tested with MacOS's bash.  `$()` does truncate the trailing newline whereas the assignment without `$()` retains it, but I don't think that it hurts anything to have that trailing newline in `$files`.  

(I keep thinking we should simply use associative arrays, and then I have to remind myself that Apple ships an ancient Bash version that doesn't support associative arrays.  Sigh.)